In [ ]:
import pandas as pd
climatic_2016 = pd.read_csv('Data/RoGovClimatic/climrbsn2016.csv')

In [2]:
climatic_2016

CODST    ALT        LAT        LON        DATCLIM  TMED  TMAX  TMIN  R24
0     15015  503.0  47.776944  23.940556  2016/01/01 00 -10.0  -5.7 -13.6  NaN
1     15015  503.0  47.776944  23.940556  2016/01/02 00 -11.0  -5.9 -14.4  NaN
2     15015  503.0  47.776944  23.940556  2016/01/03 00 -12.2  -5.5 -16.4  NaN
3     15015  503.0  47.776944  23.940556  2016/01/04 00 -11.0  -5.3 -16.6  0.0
4     15015  503.0  47.776944  23.940556  2016/01/05 00  -5.7  -4.6  -8.1  1.5
...     ...    ...        ...        ...            ...   ...   ...   ...  ...
8413  15480   12.8  44.213889  28.645556  2016/12/27 00   2.1   5.6  -0.6  0.3
8414  15480   12.8  44.213889  28.645556  2016/12/28 00   0.7   2.1  -1.5  0.4
8415  15480   12.8  44.213889  28.645556  2016/12/29 00   1.6   2.7   0.5  0.0
8416  15480   12.8  44.213889  28.645556  2016/12/30 00   0.4   1.5  -0.5  0.6
8417  15480   12.8  44.213889  28.645556  2016/12/31 00  -1.5   1.5  -2.8  0.0

[8418 rows x 9 columns]

CODST - Codul de identificare al stației meteo
ALT - Altitudine
LAT - Latitudine
LON - Longitudine
DATCLIM - Data și ora observației
TMED - Temperatura medie zilnică
TMAX - Temperatura maximă zilnică
TMIN - Temperatura minimă zilnică
R24 - Cantitatea de precipitații zilnice

In [3]:
print('Dimensiunea tabelului este de:', climatic_2016.shape)

Dimensiunea tabelului este de: (8418, 9)


### Manipularea datelor - separarea stațiilor meteo

In [4]:
# Folosim paranteze duble pentru că accesam o listă de coloane
statii_meteo = climatic_2016[['CODST', 'ALT', 'LAT', 'LON']]

In [5]:
climatic_2016 = climatic_2016.drop(columns=['ALT', 'LAT', 'LON'])

In [6]:
statii_meteo

CODST    ALT        LAT        LON
0     15015  503.0  47.776944  23.940556
1     15015  503.0  47.776944  23.940556
2     15015  503.0  47.776944  23.940556
3     15015  503.0  47.776944  23.940556
4     15015  503.0  47.776944  23.940556
...     ...    ...        ...        ...
8413  15480   12.8  44.213889  28.645556
8414  15480   12.8  44.213889  28.645556
8415  15480   12.8  44.213889  28.645556
8416  15480   12.8  44.213889  28.645556
8417  15480   12.8  44.213889  28.645556

[8418 rows x 4 columns]

In [7]:
climatic_2016

CODST        DATCLIM  TMED  TMAX  TMIN  R24
0     15015  2016/01/01 00 -10.0  -5.7 -13.6  NaN
1     15015  2016/01/02 00 -11.0  -5.9 -14.4  NaN
2     15015  2016/01/03 00 -12.2  -5.5 -16.4  NaN
3     15015  2016/01/04 00 -11.0  -5.3 -16.6  0.0
4     15015  2016/01/05 00  -5.7  -4.6  -8.1  1.5
...     ...            ...   ...   ...   ...  ...
8413  15480  2016/12/27 00   2.1   5.6  -0.6  0.3
8414  15480  2016/12/28 00   0.7   2.1  -1.5  0.4
8415  15480  2016/12/29 00   1.6   2.7   0.5  0.0
8416  15480  2016/12/30 00   0.4   1.5  -0.5  0.6
8417  15480  2016/12/31 00  -1.5   1.5  -2.8  0.0

[8418 rows x 6 columns]

In [8]:
# Eliminăm observațiile duplicate
# https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop_duplicates.html
statii_meteo = statii_meteo.drop_duplicates()
statii_meteo

CODST      ALT        LAT        LON
0     15015   503.00  47.776944  23.940556
366   15020   161.00  47.735556  26.645556
732   15090    74.29  47.163333  27.627222
1098  15108  1897.00  46.977500  25.950000
1464  15120   410.00  46.777778  23.571389
1830  15150   174.00  46.557778  26.896667
2196  15170   661.00  46.371389  25.772500
2562  15200   116.59  46.133611  21.353611
2928  15230   240.00  45.865000  22.898889
3294  15260   443.00  45.789444  24.091389
3660  15280  2504.00  45.445833  25.456667
4026  15292   241.00  45.416667  22.229167
4392  15310    69.00  45.473056  28.032222
4758  15335     4.36  45.190556  28.824167
5124  15346   237.00  45.088889  24.362778
5490  15350    97.00  45.132778  26.851667
5856  15360    12.69  45.162222  29.726944
6222  15410    77.00  44.626389  22.626111
6588  15420    90.00  44.510556  26.078056
6954  15450   192.00  44.310278  23.866944
7320  15460    18.72  44.205833  27.338333
7686  15470   102.15  44.107222  24.978611
8052  15480    12.80  44.213889  28.645556

In [9]:
statii_meteo.reset_index()

index  CODST      ALT        LAT        LON
0       0  15015   503.00  47.776944  23.940556
1     366  15020   161.00  47.735556  26.645556
2     732  15090    74.29  47.163333  27.627222
3    1098  15108  1897.00  46.977500  25.950000
4    1464  15120   410.00  46.777778  23.571389
5    1830  15150   174.00  46.557778  26.896667
6    2196  15170   661.00  46.371389  25.772500
7    2562  15200   116.59  46.133611  21.353611
8    2928  15230   240.00  45.865000  22.898889
9    3294  15260   443.00  45.789444  24.091389
10   3660  15280  2504.00  45.445833  25.456667
11   4026  15292   241.00  45.416667  22.229167
12   4392  15310    69.00  45.473056  28.032222
13   4758  15335     4.36  45.190556  28.824167
14   5124  15346   237.00  45.088889  24.362778
15   5490  15350    97.00  45.132778  26.851667
16   5856  15360    12.69  45.162222  29.726944
17   6222  15410    77.00  44.626389  22.626111
18   6588  15420    90.00  44.510556  26.078056
19   6954  15450   192.00  44.310278  23.866944
20   7320  15460    18.72  44.205833  27.338333
21   7686  15470   102.15  44.107222  24.978611
22   8052  15480    12.80  44.213889  28.645556

In [10]:
statii_meteo = statii_meteo.set_index('CODST')

In [11]:
statii_meteo

ALT        LAT        LON
CODST                               
15015   503.00  47.776944  23.940556
15020   161.00  47.735556  26.645556
15090    74.29  47.163333  27.627222
15108  1897.00  46.977500  25.950000
15120   410.00  46.777778  23.571389
15150   174.00  46.557778  26.896667
15170   661.00  46.371389  25.772500
15200   116.59  46.133611  21.353611
15230   240.00  45.865000  22.898889
15260   443.00  45.789444  24.091389
15280  2504.00  45.445833  25.456667
15292   241.00  45.416667  22.229167
15310    69.00  45.473056  28.032222
15335     4.36  45.190556  28.824167
15346   237.00  45.088889  24.362778
15350    97.00  45.132778  26.851667
15360    12.69  45.162222  29.726944
15410    77.00  44.626389  22.626111
15420    90.00  44.510556  26.078056
15450   192.00  44.310278  23.866944
15460    18.72  44.205833  27.338333
15470   102.15  44.107222  24.978611
15480    12.80  44.213889  28.645556

In [12]:
nume_statii_str = '15015-Ocna Sugatag, 15020-Botosani, 15090-Iasi, 15108-Ceahlau Toaca, 15120-Cluj-Napoca, 15150-Bacau, 15170-Miercurea Ciuc, 15200-Arad, 15230-Deva, 15260-Sibiu, 15280-Varfu Omu, 15292-Caransebes, 15310-Galati, 15335-Tulcea, 15346-Ramnicu Valcea, 15350-Buzau, 15360-Sulina, 15410-Drobeta Turnu Severin, 15420-Bucuresti-Baneasa, 15450-Craiova, 15460-Calarasi, 15470-Rosiorii de Vede, 15480-Constanta'

In [13]:
nume_statii_str

'15015-Ocna Sugatag, 15020-Botosani, 15090-Iasi, 15108-Ceahlau Toaca, 15120-Cluj-Napoca, 15150-Bacau, 15170-Miercurea Ciuc, 15200-Arad, 15230-Deva, 15260-Sibiu, 15280-Varfu Omu, 15292-Caransebes, 15310-Galati, 15335-Tulcea, 15346-Ramnicu Valcea, 15350-Buzau, 15360-Sulina, 15410-Drobeta Turnu Severin, 15420-Bucuresti-Baneasa, 15450-Craiova, 15460-Calarasi, 15470-Rosiorii de Vede, 15480-Constanta'

In [14]:
#Impartim intr-o lista de stringuri
nume_statii_list = [x.split('-', 1) for x in nume_statii_str.split(', ')]


In [15]:
nume_statii_list

[['15015', 'Ocna Sugatag'],
 ['15020', 'Botosani'],
 ['15090', 'Iasi'],
 ['15108', 'Ceahlau Toaca'],
 ['15120', 'Cluj-Napoca'],
 ['15150', 'Bacau'],
 ['15170', 'Miercurea Ciuc'],
 ['15200', 'Arad'],
 ['15230', 'Deva'],
 ['15260', 'Sibiu'],
 ['15280', 'Varfu Omu'],
 ['15292', 'Caransebes'],
 ['15310', 'Galati'],
 ['15335', 'Tulcea'],
 ['15346', 'Ramnicu Valcea'],
 ['15350', 'Buzau'],
 ['15360', 'Sulina'],
 ['15410', 'Drobeta Turnu Severin'],
 ['15420', 'Bucuresti-Baneasa'],
 ['15450', 'Craiova'],
 ['15460', 'Calarasi'],
 ['15470', 'Rosiorii de Vede'],
 ['15480', 'Constanta']]

In [16]:
nume_statii = pd.DataFrame(nume_statii_list, columns=['CODST','Nume']).astype({'CODST': 'int64'}).set_index('CODST')

In [17]:
statii_meteo = statii_meteo.join(nume_statii)

In [18]:
statii_meteo

ALT        LAT        LON                   Nume
CODST                                                      
15015   503.00  47.776944  23.940556           Ocna Sugatag
15020   161.00  47.735556  26.645556               Botosani
15090    74.29  47.163333  27.627222                   Iasi
15108  1897.00  46.977500  25.950000          Ceahlau Toaca
15120   410.00  46.777778  23.571389            Cluj-Napoca
15150   174.00  46.557778  26.896667                  Bacau
15170   661.00  46.371389  25.772500         Miercurea Ciuc
15200   116.59  46.133611  21.353611                   Arad
15230   240.00  45.865000  22.898889                   Deva
15260   443.00  45.789444  24.091389                  Sibiu
15280  2504.00  45.445833  25.456667              Varfu Omu
15292   241.00  45.416667  22.229167             Caransebes
15310    69.00  45.473056  28.032222                 Galati
15335     4.36  45.190556  28.824167                 Tulcea
15346   237.00  45.088889  24.362778         Ramnicu Valcea
15350    97.00  45.132778  26.851667                  Buzau
15360    12.69  45.162222  29.726944                 Sulina
15410    77.00  44.626389  22.626111  Drobeta Turnu Severin
15420    90.00  44.510556  26.078056      Bucuresti-Baneasa
15450   192.00  44.310278  23.866944                Craiova
15460    18.72  44.205833  27.338333               Calarasi
15470   102.15  44.107222  24.978611       Rosiorii de Vede
15480    12.80  44.213889  28.645556              Constanta

In [23]:
#temperaturile maxime si minime in 2016
print('Minima inregistrata in 2016 a fost de %.1f grade' % climatic_2016.TMIN.min())

Minima inregistrata in 2016 a fost de -24.0 grade


In [24]:
#describe - raportează valori statistice pentru fiecare coloana
climatic_2016.describe()

CODST         TMED         TMAX         TMIN          R24
count   8418.000000  8418.000000  8418.000000  8418.000000  3155.000000
mean   15275.043478    10.355524    15.862236     5.842492     5.341395
std      141.501830     9.670633    10.988680     8.798955     8.610003
min    15015.000000   -22.400000   -19.700000   -24.000000     0.000000
25%    15150.000000     3.700000     7.800000     0.000000     0.500000
50%    15292.000000    10.100000    15.500000     6.100000     2.000000
75%    15410.000000    18.500000    25.500000    12.800000     6.450000
max    15480.000000    29.500000    37.800000    26.100000    92.000000

In [25]:
temp_minima = climatic_2016[climatic_2016.TMIN == climatic_2016.TMIN.min()]

In [26]:
temp_minima

CODST        DATCLIM  TMED  TMAX  TMIN  R24
3682  15280  2016/01/23 00 -22.1 -19.5 -24.0  NaN

### Temperatura minimă din 2016 a fost înregistrată pe 23 ianuarie

In [27]:
#numele stație cu join
temp_minima.join(statii_meteo, on='CODST')

CODST        DATCLIM  TMED  TMAX  TMIN  R24     ALT        LAT  \
3682  15280  2016/01/23 00 -22.1 -19.5 -24.0  NaN  2504.0  45.445833   

            LON       Nume  
3682  25.456667  Varfu Omu

Temperatura minimă a fost pe Vârfu Omu
2504 metri altitudine.

### Unde a fost temperatura minimă în afara munților, la altitudini mai mici de 1000 metri

In [28]:
statii_meteo[statii_meteo.ALT > 1000]

ALT        LAT        LON           Nume
CODST                                             
15108  1897.0  46.977500  25.950000  Ceahlau Toaca
15280  2504.0  45.445833  25.456667      Varfu Omu

In [29]:
statii_meteo_deal_campie = statii_meteo[statii_meteo.ALT < 1000]

In [30]:
statii_meteo_deal_campie

ALT        LAT        LON                   Nume
CODST                                                     
15015  503.00  47.776944  23.940556           Ocna Sugatag
15020  161.00  47.735556  26.645556               Botosani
15090   74.29  47.163333  27.627222                   Iasi
15120  410.00  46.777778  23.571389            Cluj-Napoca
15150  174.00  46.557778  26.896667                  Bacau
15170  661.00  46.371389  25.772500         Miercurea Ciuc
15200  116.59  46.133611  21.353611                   Arad
15230  240.00  45.865000  22.898889                   Deva
15260  443.00  45.789444  24.091389                  Sibiu
15292  241.00  45.416667  22.229167             Caransebes
15310   69.00  45.473056  28.032222                 Galati
15335    4.36  45.190556  28.824167                 Tulcea
15346  237.00  45.088889  24.362778         Ramnicu Valcea
15350   97.00  45.132778  26.851667                  Buzau
15360   12.69  45.162222  29.726944                 Sulina
15410   77.00  44.626389  22.626111  Drobeta Turnu Severin
15420   90.00  44.510556  26.078056      Bucuresti-Baneasa
15450  192.00  44.310278  23.866944                Craiova
15460   18.72  44.205833  27.338333               Calarasi
15470  102.15  44.107222  24.978611       Rosiorii de Vede
15480   12.80  44.213889  28.645556              Constanta

In [31]:
climatic_deal_campie = climatic_2016.join(statii_meteo_deal_campie, on='CODST', how='inner')

In [32]:
climatic_deal_campie

CODST        DATCLIM  TMED  TMAX  TMIN  R24    ALT        LAT  \
0     15015  2016/01/01 00 -10.0  -5.7 -13.6  NaN  503.0  47.776944   
1     15015  2016/01/02 00 -11.0  -5.9 -14.4  NaN  503.0  47.776944   
2     15015  2016/01/03 00 -12.2  -5.5 -16.4  NaN  503.0  47.776944   
3     15015  2016/01/04 00 -11.0  -5.3 -16.6  0.0  503.0  47.776944   
4     15015  2016/01/05 00  -5.7  -4.6  -8.1  1.5  503.0  47.776944   
...     ...            ...   ...   ...   ...  ...    ...        ...   
8413  15480  2016/12/27 00   2.1   5.6  -0.6  0.3   12.8  44.213889   
8414  15480  2016/12/28 00   0.7   2.1  -1.5  0.4   12.8  44.213889   
8415  15480  2016/12/29 00   1.6   2.7   0.5  0.0   12.8  44.213889   
8416  15480  2016/12/30 00   0.4   1.5  -0.5  0.6   12.8  44.213889   
8417  15480  2016/12/31 00  -1.5   1.5  -2.8  0.0   12.8  44.213889   

            LON          Nume  
0     23.940556  Ocna Sugatag  
1     23.940556  Ocna Sugatag  
2     23.940556  Ocna Sugatag  
3     23.940556  Ocna Sugatag  
4     23.940556  Ocna Sugatag  
...         ...           ...  
8413  28.645556     Constanta  
8414  28.645556     Constanta  
8415  28.645556     Constanta  
8416  28.645556     Constanta  
8417  28.645556     Constanta  

[7686 rows x 10 columns]

In [33]:
climatic_deal_campie[climatic_deal_campie.TMIN == climatic_deal_campie.TMIN.min()]

CODST        DATCLIM  TMED  TMAX  TMIN  R24     ALT        LAT  \
7706  15470  2016/01/21 00 -14.9  -9.6 -23.1  NaN  102.15  44.107222   

            LON              Nume  
7706  24.978611  Rosiorii de Vede

Minimul a fost înregistrat pe 21 ianuarie, la stația din Roșiorii de Vede